x-pack/legacy/plugins/snapshot_restore/public/app/lib/authorization/components/authorization_provider.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { createContext } from 'react';
+import { useLoadPermissions } from '../services';
+
+// This is wrong as we are accesing outside of the lib root folder
+// just temporary until we have our "shared ES UI" plugin in place
+import { SectionError } from '../../../components';
+
+interface Authorization {
+  isLoaded: boolean;
+  permissions: {
+    hasAllPermissions: boolean;
+    missingPrivileges: MissingPrivileges;
+  };
+}
+
+export interface MissingPrivileges {
+  [key: string]: string[] | undefined;
+}
+
+const initialValue: Authorization = {
+  isLoaded: false,
+  permissions: {
+    hasAllPermissions: true,
+    missingPrivileges: {},
+  },
+};
+
+export const AuthorizationContext = createContext<Authorization>(initialValue);
+
+interface Props {
+  permissionEndpoint: string;
+  FormattedMessage: typeof ReactIntl.FormattedMessage;
+  children: React.ReactNode;
+}
+
+export const AuthorizationProvider = ({
+  permissionEndpoint,
+  FormattedMessage,
+  children,
+}: Props) => {
+  const { error, data: permissionsData } = useLoadPermissions(permissionEndpoint);
+
+  const isLoaded = typeof permissionsData !== 'undefined';
+
+  // Note: here we have to serialize the HTTP response
+  // If we move forward with proposal, we would need to update the SR server response
+  // to implement the interface below so we don't need the serialization.
```

Comment:
the permissions contract needs to be worked through more carefully. for SR, access to the entire app requires all `cluster` privileges, and access to `Restore status` requires all `index` privileges. the implementation logic in this PR works great for that use case, but what about for more granular UI details?

for example, maybe we want to check if user has access to _just_ read snapshots/repositories and allow them overall app access, but disallow delete/edit/restore through UI if they don't have _more_ admin privileges. read and admin privileges are both `cluster` type privileges in this case, so the current implementation wouldn't allow for this type of granular checking.

btw this comment does not block this PR! it is just some thoughts, after all I was the one who structured this app's permission response this way 😄 I am working on SLM UI for 7.4, which will also live inside Snapshot and Restore. I'm sure part of that work will be adjusting the app permissions to account for this new section of the app,  so I am happy to iterate and improve the current implementation when I get there

Reply:
I updated the logic to support this use case and be able to be granular.

Now, if we require all the privileges we need to provide `cluster.*`
If we need only _some_ privileges we can provide `cluster.someSpecifiPrivilege`